[ESIMD] Update README.md (#5515)
intel/llvm: sycl/doc/extensions/experimental/SYCL_EXT_INTEL_ESIMD/README.md

third:

```cpp
    float *A = malloc_shared<float>(Size, q);
    float *B = malloc_shared<float>(Size, q);
    float *C = malloc_shared<float>(Size, q);

    for (unsigned i = 0; i != Size; i++) {
      A[i] = B[i] = i;
    }

    q.submit([&](handler &cgh) {
      cgh.parallel_for<class Test>(
        Size / VL, [=](id<1> i)[[intel::sycl_explicit_simd]] {
        auto offset = i * VL;
        // pointer arithmetic, so offset is in elements:
        simd<float, VL> va(A + offset);
        simd<float, VL> vb(B + offset);
        simd<float, VL> vc = va + vb;
        vc.copy_to(C + offset);
      });
    }).wait_and_throw();
```

In this example the lambda function passed to the `parallel_for` is marked with
a special attribute - `[[intel::sycl_explicit_simd]]`. This tells the compiler that
this kernel is a ESIMD one and ESIMD APIs can be used inside it. Here the `simd`
objects and `copy_to` intrinsics are used which are avaiable only in the ESIMD extension.
Full runnable code sample can be found on the
[github repo](https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/vadd_usm.cpp).

#### Compiling and running ESIMD code.

> Compiling and running code that uses the ESIMD extension is the same as compiler and
> running code that uses the standard SYCL:

Reviewer: ```suggestion
Code that uses the ESIMD extension can be compiled and run using the same commands
as standard SYCL:
```

To compile using the open-source Intel DPC++ compiler:

> > `$ clang++ -fsycl vadd_usm.cpp`

Author: In open-source we do have clang++
